core/src/main/java/amino/run/kernel/client/KernelClient.java

```diff
@@ -267,13 +268,10 @@ public void measureServerMetrics() {
                 /* Measure data transfer rate by sending some arbitrary data */
                 server.receiveHeartBeat(randomBytes);
                 long t3 = System.nanoTime();
-                metric.latency = (t2 - t1) / 2;
+                metric.latency = (t2 - t1); // RTT
 
-                /* TODO: T3-T2(heartbeat with data) is sometimes smaller than T2-T1(empty heartbeat). Need to relook */
                 /* Data transfer rate in Mbps */
-                metric.rate =
-                        ((RANDOM_DATA_SIZE * 8 * 1000.0 * 1000.0)
-                                / (((t3 - t2) - (t2 - t1)) * 1024.0 * 1024.0));
+                metric.rate = (8.0 / DATA_SIZE_IN_MB) / ((t3 - t2)/(1000.0 * 1000.0 * 1000.0));
```

Comment:
This is wrong, or at best very confusing.  You want bytes per second.

So:
```
 DATA_SIZE_IN_BYTES/(t3 - t2)/TimeUnit.SECONDS.toNanos(1);
```
?
Of course, beware of integer arithmetic.

Reply:
Unit is in Mbps(MegaBits/Sec)